test/spec.credentials-helper.js

```diff
@@ -11,26 +11,33 @@ describe('credentials-helper', () => {
     const remote = 'origin';
     const payload = { username: 'testuser', password: 'testpassword' };
     const server = http.createServer((req, res) => {
-      const reqUrl = url.parse(req.url);
-      expect(reqUrl.pathname).to.be('/api/credentials');
+      try {
+        const reqUrl = url.parse(req.url);
+        expect(reqUrl.pathname).to.be('/api/credentials');
 
-      const params = querystring.parse(reqUrl.query);
-      expect(params['remote']).to.be(`${remote}`);
-      expect(params['socketId']).to.be(`${socketId}`);
+        const params = querystring.parse(reqUrl.query);
+        expect(params['remote']).to.be(`${remote}`);
+        expect(params['socketId']).to.be(`${socketId}`);
 
-      res.writeHead(200, { 'Content-Type': 'application/json' });
-      res.end(JSON.stringify(payload));
+        res.writeHead(200, { 'Content-Type': 'application/json' });
+        res.end(JSON.stringify(payload));
+      } finally {
+        if (!res.finished) {
```

Comment:
It seems that `finished` is deprecated: https://nodejs.org/api/http.html#http_response_finished
Maybe this should use `writableEnded` or `writableFinished` instead?

Reply:
These two APIs were added in `v12.9.0` and `v12.7.0` so we have to use `finished` for now...

Reply:
Oh, I didn't see that part, I was too distracted by the big, red "Deprecated" banner. 😄